Back end signup / login api routes

quddd opened the pull request.

quddd committed:
Create profileModel.js

Initial commit for BE: Profile Model

quddd committed:
Merge pull request #43 from hatchways/BE-Profile-Model

BE: Profile Model

dproc96 reviewed (dismissed):
I'm guessing this PR is because when you merged the Profile Model branch it merged into this branch rather than dev, is that correct? You can avoid that in the future like this:

If you have a PR to merge branch B into branch A and one to merge branch A into dev, when you merge branch A into dev an option should come up on github saying "This branch can be safely deleted". If you click "Delete Branch", then branch B's PR will automatically switch from branch A to dev.

quddd committed:
Create profileControllers.js

quddd committed:
Finished profileControllers

dproc96 commented on server/controllers/profileControllers.js:
```diff
@@ -0,0 +1,76 @@
+exports.updateProfile = async (req, res, next) => {
+    try{
+        //retrieve profile and update necessary fields
+        const updatedProfile = await Profile.findOneAndUpdate(
```
It would be a good idea to add a 404 error if no profile is found (like you have for the GET route)

dproc96 commented on server/controllers/profileControllers.js:
```diff
@@ -0,0 +1,76 @@
+    try{
+        const profile = await Profile.findById(req.params.id);
+        if (!profile){
+            return next(createError(400, "Profile does not exist!"));
```
```suggestion
            return next(createError(404, "Profile does not exist!"));
```
This would be a better status code to return in this case

quddd committed:
update userRouter.js

removed profileController from userRouter

dproc96 commented on server/routes/userRouter.js:
```diff
@@ -1,6 +1,7 @@
 const router = require("express").Router();
 
 const authController = require("../controllers/authController");
+const profileController = require("../controllers/profileControllers")
```
Would probably be better to have a router for the profile outside of the user router

dproc96 reviewed (commented):
Overall this looks great, just a few small changes I noticed

quddd committed:
Added 404 error for invalid id's and id's that don't exist

Added 404 error for invalid id's and id's that don't exist to profileControllers.js

quddd committed:
change status code 400 to 404 for profiles that are not found

changed 400 to 404 in profileControllers.js for profiles not found

dproc96 reviewed (commented):
This looks good, but to test it you'll need to hook it into a profile Router

quddd committed:
created profile router and made routes protected

created profile router and made profile routes protected apart from create route

dproc96 reviewed (approved):
Everything is working locally, this is good to go!
